lib/find.ts

```diff
@@ -1,32 +1,42 @@
 import * as fs from 'fs';
 import { join } from 'path';
-import { promisify } from 'util';
 import { debug } from './debug';
 
-const readdir = promisify(fs.readdir);
-const stat = promisify(fs.stat);
+const MAX_FILE_SIZE: number = 2 * 1024 * 1024 * 1024;
 
-export async function find(dir: string): Promise<string[]> {
-  const result: string[] = [];
+export interface FsHandler {
+  stat(path: fs.PathLike): fs.Stats;
+  readdir(path: string): string[];
+}
+
+const defaultFsHandler: FsHandler = {
+  stat: fs.statSync,
+  readdir: fs.readdirSync,
+};
 
-  const dirStat = await stat(dir);
+export function find(
+  dir: string,
+  fsHandler: FsHandler = defaultFsHandler,
+): string[] {
+  const result: string[] = [];
+  const dirStat = fsHandler.stat(dir);
```

Comment:
why we moved from promisify.fs.stat to statSync?

Reply:
this was an inherit from my changes made in #49. I don't think it's necessary in this PR, and maybe not even in #49, I need to verify if so still is the case.